cuda_core/docs/source/interoperability.rst

```diff
@@ -0,0 +1,83 @@
+.. currentmodule:: cuda.core.experimental
+
+Interoperability
+================
+
+``cuda.core`` is designed to be interoperable with other Python GPU libraries. Below
+we cover a list of possible such scenarios.
+
+
+Current device/context
+----------------------
+
+The :meth:`Device.set_current` method ensures that the calling host thread has
+an active CUDA context set to current. This CUDA context can be seen and accessed
+by other GPU libraries without any code change. For libraries built on top of
+the CUDA runtime (``cudart``), this is as if ``cudaSetDevice`` is called.
+
+Since CUDA contexts are per-thread constructs, in a multi-threaded program each
+host thread should call this method.
+
+Conversely, if any GPU library already set a device (or context) to current, this
+method ensures that the same device/context is picked up by and shared with
+``cuda.core``.
+
+
+``__cuda_stream__`` protocol
+----------------------------
+
+The :class:`~_stream.Stream` class is a vocabulary type representing CUDA streams
+in Python. While we encourage new Python projects to start using streams (and other
+CUDA types) from ``cuda.core``, we understand that there are already several projects
+exposing their own stream types.
+
+To address this issue, we propose the ``__cuda_stream__`` protocol (currently version
+0) as follows: For any Python objects that are meant to be interpreted as a stream, they
+should add a ``__cuda_stream__`` attribute that returns a 2-tuple: The version number
+(``0``) and the address of ``cudaStream_t``:
```

Comment:
We should clarify here that the address is a Python `int`.

Reply:
Noted that both are Python int in c06270c92b4a779314a029ac75c1645ab242b00c.